sympy/polys/polytools.py

```diff
@@ -1818,6 +1818,30 @@ def total_degree(f):
         else:  # pragma: no cover
             raise OperationNotSupported(f, 'total_degree')
 
+    def power_list(f):
+        """
+        Returns a list of powers of variables in ``f``.
+
+        Examples
+        ========
+
+        >>> from sympy import Poly
+        >>> from sympy.abc import x, y
+
+        >>> Poly(x**2 + y*x + 1, x, y).power_list()
+        (2, 1, 0)
+        >>> Poly(0, x).power_list()
+        (0,)
+        >>> Poly(1, x).power_list()
+        (0,)
+
+        """
+        power = []
+        for i in f.monoms():
+            power += list(i)
+        power = reversed(list(set(power)))
+        return tuple(power)
```

Comment:
How will the user make sense of the result when there is more than one generator? If you look at `degree` I think you will see how we tried to handle that situation there. I think the approach is to allow no generator to be specified when the expression is univariate but to raise an error if no generator is given for the multivariate case.

Reply:
Okay I get your point. I will make this function work only when then generator is univariate. Can you please suggest how could I check if it is univariate and what kind of error should be raised.